client/src/templates/Challenges/components/scene/scene.tsx

```diff
@@ -21,11 +21,32 @@ export function Scene({ scene }: { scene: FullScene }): JSX.Element {
     new Audio(`${sounds}/${audio.filename}${audioTimestamp}`)
   );
 
+  const loadImage = (src: string) => {
+    const img = new Image();
+    img.src = src;
+  };
+
   // on mount
   useEffect(() => {
     const { current } = audioRef;
     current.addEventListener('canplaythrough', audioLoaded);
 
+    // preload images
+    loadImage(`${backgrounds}/${setup.background}`);
+
+    setup.characters.forEach(character => {
+      const characterImages = characterAssets[character.character];
+      Object.values(characterImages).forEach(src => {
+        if (src) {
+          loadImage(src);
+        }
+      });
+    });
```

Comment:
```suggestion
    setup.characters
      .map(({ character }) => Object.values(characterAssets[character]))
      .flat()
      .forEach(loadImage);
```
The linear pipeline is easier to read, imo.